hosts/galileo/__init__.py

```diff
@@ -0,0 +1,395 @@
+    @staticmethod
+    def reset():
+        """Reset the internal parameters.
+
+        Can be useful for debugging.
+        """
+        Galileo.loaded_instruments = []
+
+        Galileo.CK_LOADED = np.zeros(Galileo.MONTHS, dtype='bool')
+        Galileo.CK_LIST = np.empty(Galileo.MONTHS, dtype='object')
+        Galileo.CK_DICT = {}
+
+        Galileo.SPK_LOADED = np.zeros(Galileo.MONTHS, dtype='bool')
+        Galileo.SPK_LIST = np.empty(Galileo.MONTHS, dtype='object')
+        Galileo.SPK_DICT = {}
+
+        Galileo.initialized = False
+
+    #===========================================================================
+    @staticmethod
+    def load_ck(t):
+        """Ensure that the C kernels applicable at or near the given time have
+        been furnished.
+
+        The time can be tai or tdb.
+        """
+        Galileo.load_kernels(t, t, Galileo.CK_LOADED, Galileo.CK_LIST,
+                                   Galileo.CK_DICT)
+
+    #===========================================================================
+    @staticmethod
+    def load_cks(t0, t1):
+        """Ensure that all the C kernels applicable near or within the time
+        interval tdb0 to tdb1 have been furnished.
+
+        The time can be tai or tdb.
+        """
+        Galileo.load_kernels(t0, t1, Galileo.CK_LOADED, Galileo.CK_LIST,
+                                     Galileo.CK_DICT)
+
+    #===========================================================================
+    @staticmethod
+    def load_spk(t):
+        """Ensure that the SPK kernels applicable at or near the given time have
+        been furnished.
+
+        The time can be tai or tdb.
+        """
+        Galileo.load_kernels(t, t, Galileo.SPK_LOADED, Galileo.SPK_LIST,
+                                   Galileo.SPK_DICT)
+
+    #===========================================================================
+    @staticmethod
+    def load_spks(t0, t1):
+        """Ensure that all the SPK kernels applicable near or within the time
+        interval tdb0 to tdb1 have been furnished.
+
+        The time can be tai or tdb.
+        """
+        Galileo.load_kernels(t0, t1, Galileo.SPK_LOADED, Galileo.SPK_LIST,
+                                     Galileo.SPK_DICT)
+
+    #===========================================================================
+    @staticmethod
+    def load_kernels(t0, t1, loaded, lists, kernel_dict):
+        from oops.unittester_support import TESTDATA_PARENT_DIRECTORY
+        import glob
+
+        kdir = os.path.join(TESTDATA_PARENT_DIRECTORY, '../SPICE/Galileo/')
+        gkdir = os.path.join(TESTDATA_PARENT_DIRECTORY, '../SPICE/General/')
+
+        cspyce.furnsh(gkdir + 'LSK/naif0012.tls')
+        cspyce.furnsh(kdir + 'SCLK/mk00062a.tsc')
+        cspyce.furnsh(kdir + 'IK/gll36001.ti')
+        cspyce.furnsh(kdir + 'SPK/de421.bsp')
+        cspyce.furnsh(kdir + 'SPK/de432s.bsp')
+
+        [cspyce.furnsh(ckfile) for ckfile in glob.glob(kdir + 'CK/*.bc')]
```

Comment:
I'm not fond of having a list comprehension with side effects. I think this would be a lot more readable as a for loop.

Reply:
fixed